### ISSUE TITLE
FLASK_APP variable isn't behaving as specified.

### ISSUE THREAD

**Issue by doomcat55** [pallets/flask#1970]
[The docs](http://flask.pocoo.org/docs/0.11/cli/) on the FLASK_APP variable state that

> In that imported file the name of the app needs to be called `app` or optionally be specified after a colon. For instance _mymodule:application_ would tell it to use the _application_ object in the `mymodule.py` file.

However, creating the following script (run.py):
```
from flask import Flask

webapp = Flask(__name__)
```
setting `FLASK_APP=run.py:webapp`, and running `flask run` gives me an ImportError:
```
Traceback (most recent call last):
  File "/Users/me/app/venv/lib/python3.5/site-packages/flask/cli.py", line 156, in __call__
    self._flush_bg_loading_exception()
  File "/Users/me/app/venv/lib/python3.5/site-packages/flask/cli.py", line 144, in _flush_bg_loading_exception
    reraise(*exc_info)
  File "/Users/me/app/venv/lib/python3.5/site-packages/flask/_compat.py", line 33, in reraise
    raise value
  File "/Users/me/app/venv/lib/python3.5/site-packages/flask/cli.py", line 133, in _load_app
    self._load_unlocked()
  File "/Users/me/app/venv/lib/python3.5/site-packages/flask/cli.py", line 148, in _load_unlocked
    self._app = rv = self.loader()
  File "/Users/me/app/venv/lib/python3.5/site-packages/flask/cli.py", line 209, in load_app
    rv = locate_app(self.app_import_path)
  File "/Users/me/app/venv/lib/python3.5/site-packages/flask/cli.py", line 89, in locate_app
    __import__(module)
ImportError: No module named 'run'
```
Interestingly, using `FLASK_APP=run.py` actually works even with the app named "webapp". So while I'm not sure how much of this is error on my part vs. error on flask's, the docs are wrong in saying that the Flask object _needs_ to be named "app".

**Comment by wgwz** [comment]
This works for me: `export FLASK_APP=run` or `export FLASK_APP=run:webapp` and `python -m flask run`. It looks like the base issue is #1847. But it is fixed by #1872? 

- Also related: #1829, #1902
- Distantly related: #1926

**Comment by doomcat55** [comment]
Reading the links, seems the issue is that the package needs to be installed for the CLI to work right. It would be nice if it that wasn't necessary, but I'm closing this because it's a duplicate of [#1847](https://github.com/pallets/flask/issues/1847). Thanks!
